external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -346,6 +346,31 @@ pub enum ManagedAddressUpdate {
     },
 }
 
+/// Sets the `original-published-id` to a given `id` in the lock file. This is a raw utility
+/// for preparing package publishing and package upgrades. Invariant: callers maintain a valid
+/// hex `id`.
+pub fn set_original_id(file: &mut LockFile, environment: &str, id: &str) -> Result<()> {
```

Comment:
We discussed an in-memory representation of lock contents, and this kind of thing makes me eager to make that work instead of using the file. An in-memory representation needs to be a separate refactor though, and I think potentially really error-prone based on how things happen right now. So I went with the predictable, mechanical way of working via the lock file for now.